Dockerfile

```diff
@@ -15,44 +15,40 @@
 # Build the manager binary
 FROM golang:1.20 as builder
 
-ARG ARCH ?= $(shell go env GOARCH)
-ARG TARGETPLATFORM
-ENV TARGETPLATFORM=${TARGETPLATFORM:-linux/amd64}
-ARG LDFLAGS
-ARG GO_BUILD_GC_ARGS
-
 WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg/mod \
+    go mod download -x
 
 # Copy the go source
-COPY main.go main.go
-COPY api/ api/
-COPY controllers/ controllers/
-COPY pkg/ pkg/
-COPY version/ version/
+COPY ./ ./
 
-# Add kubectl tool
-RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/${TARGETPLATFORM}/kubectl"
-RUN chmod +x ./kubectl
```

Comment:
This wasn't caching - so removing it makes image builds much faster as they don't include the download.

I'm not sure where the image is currently build that might be missing this.

Reply:
Need to sync with @e0ne .
This is not matching with this:
https://gitlab-master.nvidia.com/cloud-orchestration/cicd/kubernetes-ci/-/merge_requests/118/diffs

Reply:
I've removed this part as it wasn't reponsible for that much of a speed up. It's network dependent so it's highly variable. With the go cache in use it accounts for at least half of the build time. I think we should remove it if possible, but I'm not sure of the consequences.